src/model/Model.js

```diff
@@ -57,6 +57,7 @@ export default class Model {
 		const adaptors = this.root.adaptors;
 		const value = this.value;
 		const len = adaptors.length;
+		if ( len === 0 ) return;
```

Comment:
Being a little nitpicky, might want to pull this up just after `adaptors`. The `value` assignment becomes useless if it bails out early.

Might also want to look at those 2 hacks down below. It seems your performance boost relies on avoiding `getKeypath`. If those comments tell the truth, there could be another way of doing this function, avoiding all the hacks in this sector of the code. :smile: 

Other than that, :+1: